### ISSUE TITLE
Suggestion: On-screen feedback for parry training and throw teching

### ISSUE THREAD

**Issue by rsnoob35** [Grouflon/3rd_training_lua#12]
If a parry attempt is detected within, say, 10 frames before or after the parry window for a move starts or ends, a popup would tell you exactly how early/late you were. If you landed the parry, it would tell you what frame of the parry window the move connected on (for example, 1/5 would come up if your parry came out on the same frame a jump in attack connected). This would help newer players get the timing down for regular parries and help more advanced players practice red parries.

In the same vein as this, it would be nice to have feedback for which frame of the throw tech window you input tech on. Right now, you can only practice late teching by using replays, and it's impossible to tell if the late tech you input would've been hit by a move that hits you for doing the tech very slightly too early

**Comment by Grouflon** [comment]
That would be great yeah.
Do you have any viable info on what are the time frames for parrying, red parrying and teching?

**Comment by rsnoob35** [comment]
High and low parry, respectively, are both active for 10 frames. Once input, another parry in the same direction cannot be input for 23 frames, counting the first frame of parry as frame 1. Your earliest second parry attempt can be input on frame 24, and becomes active on frame 25.. 

Jumping parry is active for 7 frames. Once input, another air parry cannot be input for 20 frames, counting the first frame of parry as frame 1. Your earliest second parry attempt can be input on frame 21, and becomes active on frame 22.

Anti-air parry is active for 5 frames. Once input, another air parry cannot be input for 18 frames, counting the first frame of parry as frame 1. Your earliest second parry attempt can be input on frame 19, and becomes active on frame 20.

Parry attempts during the lockout window do not reset the 
parry lockout timer.

Red parrying a normal attack is a 3 frame window, red parrying a special or super is a 2 frame window.  

The stick must return to neutral for at least 1 frame before attempting a parry of any sort.

The parry values above assume you tap the stick when inputting a parry. If you continue holding the stick in the direction of the parry, the game will truncate the parry window to 6 frames, provided the parry window was greater than 6 to begin with. This doesn't affect anti-air parries or red parries. 

There is a 5 frame window to tech throws after they have connected, inclusive of the frame the throw connects. If an enemy throw connects on frame 2 or 3 of your own throw, you will also get a tech. However, an enemy throw connecting on frame 1 of your own throw will not be teched. Moreover, this counts as getting counterhit out of the startup of a standard move, which means you CANNOT tech the throw, even if you were to double tap tech fast enough to hit the later tech window.

**Comment by Grouflon** [comment]
That's good info, thanks !
Aboute truncated parry windows, what would be considered not tapping the stick then ? just an above 1 frame input ?

**Comment by rsnoob35** [comment]
Right, sorry, the game checks on frame 5 of parry to see if the stick is still held or not

**Comment by rsnoob35** [comment]
In the same vein as throw teching, maybe it would be useful to show how many frames the player was blocking/parrying/standing still for before doing another action. This would be useful to practice guard jumping, or other defensive tech that generally involves precise delays

**Comment by abuchachos** [comment]
Just want to add that all of the data @rsnoob35 mentioned is correct, but only applies to unsuccessful parry attempts (sorry, I'm sure this is obvious to most. I'm just stating the obvious for the sake of clarity).

If a parry is successful, there is no lockout period and you can parry again immediately (please correct me if I am wrong)
